packages/blade/src/components/Input/BaseInput/BaseInput.tsx

```diff
@@ -865,7 +868,7 @@ export const BaseInput = React.forwardRef<BladeElementRef, BaseInputProps>(
               inputWrapperRef={inputWrapperRef}
             >
               <StyledBaseInput
-                as={isReactNative ? undefined : as}
+                as={as}
```

Comment:
Won't this break RN if someone passes `as`?

Reply:
it is handled inside `StyledBaseInput`. We do want to let `as` pass here so that depending on as="button" vs as="input" we can render the correct RN element.